### PR TITLE
Testing Restructure and V2 Update

### PR DESCRIPTION
**Pull request by zacherysmith115**
Added functionality to: 
- Set up session start, session end, and fixtures using pytest 
- Implemented the mock classes for QAModelDTOs, DataDTOs, and VisitorDTOs 
- Rewrote test cases to use the pytest fixtures and v2 implementations 
- Added the missing test cases for DataServices and QAModelService
- Added a few end points to ensure all test cases are working for QAModelController

### CODE REVIEW

File: controllers/v2/model_controller_v2.py [hunk 1]
```diff
@@ -31,14 +31,34 @@ def __init__(
         self.data_service = data_service
 
     def get_models_by_type(self, model_type: str) -> List[QAModelResponseV2]:
-        models: List[QAModel] = self.service.read_all_qa_model_by_type(model_type)
+        models: List[QAModel] = self.service.read_all_qa_model_by_type(
```
Comment (edwardlach) [inline controllers/v2/model_controller_v2.py:34]: Any reason for the new line in this function call?  Seems like a one liner would make more sense.

File: controllers/v2/model_controller_v2.py [hunk 2]
```diff
@@ -31,14 +31,34 @@ def __init__(
...
+
+        if len(models) > 0:
+            return [QAModelResponseV2.from_model(model) for model in models], 200
+        else:
+            return None, 200
+
     def get_word_cloud(self, resource_id: str) -> WordCloudResponseV2:
-        word_cloud: List[WordResponse] = self.data_service.generate_word_cloud(resource_id)
+        word_cloud: List[WordResponse] = self.data_service.generate_word_cloud(
+            resource_id)
```
Comment (edwardlach) [inline controllers/v2/model_controller_v2.py:61]: Same here, any reason for the new line?

File: dtos/v2/prediction_dto_v2.py [hunk 3]
```diff
@@ -52,3 +52,51 @@ def from_model(model: Prediction):
...
+#         return Prediction(prediction=self.datum.answer,
+#                           model_id=self.model_id,
+#                           datum=self.datum.to_model(),
+#                           visitor_id=self.visitor.id)
+
+
+# @dataclass
+# class PredictionUpdateRequestV2(object):
+#     id: int
+#     is_correct: bool
```
Comment (edwardlach) [inline dtos/v2/prediction_dto_v2.py:74]: I think you are good to uncomment all of these dtos, they should be valid

File: models/visitor_model.py [hunk 4]
```diff
@@ -7,7 +7,6 @@ class Visitor(IdentifiableEntity):
     __tablename__ = 'visitor'
 
     token = db.Column(String(60), nullable=False, unique=True)
-    #name = db.Column(String(60), nullable=False)  Not used as per latest diagram - CSJ 2022-01-26
```
Comment (edwardlach) [inline models/visitor_model.py:10]: Should be good to just remove this then, no need to comment  it out.

File: tests/conftest.py [hunk 5]
```diff
@@ -1,31 +1,109 @@
...
+def data_service():
+    service = DataService()
+    yield service
+
+
+@pytest.fixture
+def qa_model_model(db: SQLAlchemy):
+
+    dto: QAModelCreateRequestV2 = MockQAModelCreateRequestV2()
+    qa_model = dto.to_model()
```
Comment (edwardlach) [inline tests/conftest.py:64]: Should probably just do this in one line, qa_model: QAModel = MockQAModelCreateRequestV2().to_model()

File: tests/conftest.py [hunk 6]
```diff
@@ -1,31 +1,109 @@
...
+@pytest.fixture
+def qa_model_service():
+    service = QAModelService()
+    yield service
+
+
+@pytest.fixture
+def visitor_model(db: SQLAlchemy):
+    dto: VisitorCreateRequestV2 = MockVisitorCreateRequestV2()
+    visitor_model = dto.to_model()
```
Comment (edwardlach) [inline tests/conftest.py:93]: Same note as above, should probably just be one line

**Review by edwardlach — APPROVED** [review]
Damn Zach, you were on a testing tear!  Looks really good, my comments were all small and nitpicky.

File: controllers/v2/model_controller_v2.py (hunk 1 above)
Comment (zacherysmith115) [reply controllers/v2/model_controller_v2.py:34]: Auto formatter probably, first time messing with it.

File: models/visitor_model.py (hunk 4 above)
Comment (zacherysmith115) [reply models/visitor_model.py:10]: Was removed.

**Pull request merged**